#### Урок 7. Непараметрические тесты

1-4 задачи решать с помощью функций, 5ю вручную

Выбрать тест и проверить, есть  ли различия между выборками:

1.  Даны две  независимые выборки. Не соблюдается условие нормальности

x1  380,420, 290

y1 140,360,200,900

Сделайте вывод по результатам, полученным с помощью функции

In [2]:
import numpy as np
from scipy import stats 

In [3]:
x1 = np.array([380,420, 290]) 
y1 = np.array([ 140,360,200,900]) 

In [4]:
stats.mannwhitneyu(x1,y1)

MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

Так как p-value (62.8%) > a (5%), статистических отличий между выборками не наблюдается 

2. Исследовалось влияние препарата на уровень давления пациентов. Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут. Есть ли статистически значимые различия?

1е измерение до приема препарата: 150, 160, 165, 145, 155

2е измерение через 10 минут: 140, 155, 150,  130, 135

3е измерение через 30 минут: 130, 130, 120, 130, 125

In [5]:
a = np.array([150, 160, 165, 145, 155])
b = np.array([140, 155, 150,  130, 135])
c = np.array([130, 130, 120, 130, 125])

In [6]:
stats.friedmanchisquare(a,b,c)

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

Так как p-value (0.8%) < a (5%), наблюдаются статистические отличия между выборками. Таким образом принимаем альтернативную гипотезу, что препарат влияет на уровень давления пациентов

3. Сравните 1 и 2 е измерения, предполагая, что 3го измерения через 30 минут не было.

In [7]:
stats.wilcoxon(a,b)

WilcoxonResult(statistic=0.0, pvalue=0.0625)

Так как p-value (6.25%) > a (5%), статистических отличий между выборками не наблюдается. Таким образом делаем вывод, что препрать не повлиял на уровен давления пациентов.

4. Даны 3 группы  учеников плавания.

В 1 группе время на дистанцию 50 м составляют: 56, 60, 62, 55, 71, 67, 59, 58, 64, 67

Вторая группа : 57, 58, 69, 48, 72, 70, 68, 71, 50, 53

Третья группа: 57, 67, 49, 48, 47, 55, 66, 51, 54

In [10]:
a = np.array([56, 60, 62, 55, 71, 67, 59, 58, 64, 67])
b = np.array([57, 58, 69, 48, 72, 70, 68, 71, 50, 53])
c = np.array([57, 67, 49, 48, 47, 55, 66, 51, 54])

In [11]:
stats.kruskal(a,b,c)

KruskalResult(statistic=5.465564058257224, pvalue=0.06503809985904942)

Так как p-value (6.5%) > a (5%), статистических отличий между выборками не наблюдается. Таким образом делаем вывод, что разниц между группами учеников нет

5. Заявляется, что партия изготавливается со средним арифметическим 2,5 см. Проверить данную гипотезу, если известно, что размеры изделий подчинены нормальному закону распределения. Объем выборки 10, уровень статистической значимости 5%

2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34

In [16]:
x = np.array([2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34])
n = 10
s = np.std(x, ddof=1)
mu = 2.5
average_x = np.mean(x)

In [14]:
t = (mu-average_x)/(s/n**0.5)
t

-0.5630613661802959

In [15]:
t_tabl = stats.t.ppf(0.975, n-1)
t_tabl

2.2621571627409915

In [17]:
t_tabl = stats.t.ppf(0.025, n-1)
t_tabl

-2.262157162740992

Так как -0.56 находится на между -2,26 и -2.26, то нулевая гипотеза верна и среднее арифметическое партии равно 2,5 см

Проверка по функции, также дает принятие нулевой гипотезы

In [18]:
stats.ttest_1samp(x,mu)

TtestResult(statistic=0.5630613661802959, pvalue=0.5871439993940629, df=9)